test/services/submission/SubscriptionBlockBuilderSpec.scala

```diff
@@ -255,6 +283,23 @@ class SubscriptionBlockBuilderSpec extends VatRegSpec with VatRegistrationFixtur
       result mustBe minimalSubscriptionBlockJson
     }
 
+    "build a minimal subscription json when minimum data is provided and user is an intending trader" in {
+      val vatScheme = testVatScheme.copy(
+        applicantDetails = Some(validApplicantDetails),
+        vatApplication = Some(testVatApplicationDetails.copy(appliedForExemption = Some(true))),
+        eligibilitySubmissionData = Some(testEligibilitySubmissionData.copy(
+          threshold = Threshold(mandatoryRegistration = false, None, None, None),
+          registrationReason = IntendingTrader
```

Comment:
We seem to be testing the old state of intending trader reason coming from eligibility. It would be ideal to test it against the `currentlyTrading` flat in `VatApplication` instead, which is more relevant to this story. Thanks